frontend/src/modules/Shares/components/ShareLogs.js

```diff
@@ -0,0 +1,119 @@
+import Editor from '@monaco-editor/react';
+import { RefreshRounded } from '@mui/icons-material';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  Grid,
+  Typography
+} from '@mui/material';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState } from 'react';
+import { THEMES, useSettings } from 'design';
+import { SET_ERROR, useDispatch } from 'globalErrors';
+import { getShareLogs, useClient } from 'services';
+
+export const ShareLogs = (props) => {
+  const { shareUri, onClose, open } = props;
+  const { settings } = useSettings();
+  const client = useClient();
+  const dispatch = useDispatch();
+  const [logs, setLogs] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  const getLogs = useCallback(async () => {
+    setLoading(true);
+    try {
+      const response = await client.query(getShareLogs(shareUri));
+      if (response && !response.errors) {
+        setLogs(response.data.getShareLogs.map((l) => l.message));
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    }
+    setLoading(false);
+  }, [client, dispatch, shareUri]);
+
+  useEffect(() => {
+    if (client && open) {
+      getLogs().catch((e) => dispatch({ type: SET_ERROR, error: e.message }));
+    }
+  }, [client, dispatch, getLogs]);
```

Comment:
add `open` to the dependency list here - otherwise `getLogs()` not running when `Logs` Button clicked on via UI (needed to Refresh once to invoke API beforehand)

Reply:
Done